### PR TITLE
feat(ui-shell): switch header nav role for side nav

### LINKED ISSUE #3572: AVT 1 - React Header Base w/Navigation or the Header Base w/Navigation and Actions UI Shell has DAP violations
## Environment
macOS Mojave version 10.14.5
Chrome Version 75.0.3770.100 (Official Build) (64-bit)
Carbon v10 - React
DAP - July 2019 Ruleset

## Detailed Description
Run DAP on the Header Base w/Navigation or the Header Base w/Navigation and Actions UI Shell. The following 5 violations are identified (see screenshot below):
HeaderNav

Note: 
1. The contrast issue relates to the Skip to Main link, which fails the WCAG Tests for [G1: Adding a link at the top of each page that goes directly to the main content area](https://www.w3.org/TR/WCAG20-TECHS/G1.html#G1-examples)
2. Add a role to the parent  of the menuitems. Use role="menu" or "menubar" or "group"  to fix the issue. DAP only reports this issue if the page size reduced and the links do not display across the top of the page.

### PR DESCRIPTION
**Pull request by asudoh**
This change allows users of `<HeaderMenu>` and `<HeaderMenuItem>` to specify their ARIA roles so the a11y structure won't be broken when they are put in side nav for narrow screen.

Fixes #3572.

#### Changelog

**New**

- `linkRole` prop to `<HeaderMenu>` and `<HeaderMenuItem>`, that allows to specify their ARIA roles of the `<a>` in them.

#### Testing / Reviewing

Testing should make sure `<HeaderMenu>` and `<HeaderMenuItem>` are not broken.

### CODE REVIEW

**Review by joshblack — CHANGES_REQUESTED** [review]
What issue is this corresponding to btw? If I remember correctly, we ended up just removing the menu pattern so should we only remove `role="menuitem"` and leave it at that? In particular, `role="menuitem"` doesn't work as intended since we removed the parent `role="menu"` (or was that in another component? 🤔)

**Comment by asudoh** [comment]
Removing `role="menuitem"` sounds fine to me, updated.

**Review by joshblack — APPROVED** [review]

**Comment by joshblack** [comment]
Seems great 🎉 just need to have merge commit be something like:

```git
fix(ui-shell): remove role="menuitem" from header links
```

**Comment by abbeyhrt** [comment]
@asudoh will this address the "An element with WAI-ARIA role must be contained within a valid element" violation in #3576 as well?

**Comment by abbeyhrt** [comment]
Screen Shot 2019-12-13 at 12 22 06 PM
Not sure what's happening but something is going on with the Links now

**Comment by joshblack** [comment]
Ah good catch @abbeyhrt 🤦‍♂️ forgot that the UI Shell selectors are tied to the ARIA role for `menuitem`

**Comment by abbeyhrt** [comment]
Hey @asudoh! Any progress on this?

**Comment by asudoh** [comment]
My apologies @abbeyhrt, this completely slipped my mind. Fixed.

> @asudoh will this address the "An element with WAI-ARIA role must be contained within a valid element" violation in #3576 as well?

Not an intent, but seems that you are right (as I ran the DAP).

File: packages/components/src/components/ui-shell/_side-nav.scss [hunk 1]
```diff
@@ -643,28 +643,28 @@
       width: 100%;
     }
 
-    & a.#{$prefix}--header__menu-item[role='menuitem'] {
+    & a.#{$prefix}--header__menu-item {
```
Comment (emyarod) [inline packages/components/src/components/ui-shell/_side-nav.scss]:
```suggestion
    a.#{$prefix}--header__menu-item {
```

File: packages/components/src/components/ui-shell/_side-nav.scss [hunk 2]
```diff
@@ -643,28 +643,28 @@
...
       padding-left: 4.25rem;
       font-weight: 400;
     }
 
-    & a.#{$prefix}--header__menu-item[role='menuitem']:hover {
+    & a.#{$prefix}--header__menu-item:hover {
```
Comment (emyarod) [inline packages/components/src/components/ui-shell/_side-nav.scss]:
```suggestion
    a.#{$prefix}--header__menu-item:hover {
```

**Review by emyarod — APPROVED** [review]
looks good to me, just removing redundant sass prefix

**Comment by abbeyhrt** [comment]
@asudoh this looks great! Just one thing I noticed, I'm not sure if this is just because of changes to the story but the `Link 4` sub menu won't open for me in the Header Base w/ Navigation story

**Comment by asudoh** [comment]
Great catch @abbeyhrt, my apologies for leaving another issue. Fixed.

**Review by abbeyhrt — APPROVED** [review]
Looks great! 🎉

**Pull request merged**